lineplot/lineplot.app.js

```diff
@@ -0,0 +1,231 @@
+(function () {
+    'use strict';
+
+    angular.module('chaise.configure-lineplotApp', ['chaise.config'])
+
+        .constant('appName', 'lineplotApp')
+
+        .run(['$rootScope', function ($rootScope) {
+            // When the configuration module's run block emits the `configuration-done` event, attach the app to the DOM
+            $rootScope.$on("configuration-done", function () {
+
+                angular.element(document).ready(function () {
+                    angular.bootstrap(document.getElementById("lineplot"), ["lineplotApp"]);
+                });
+            });
+        }]);
+
+        angular.module('lineplotApp', [
+            'ngSanitize',
+            'ngCookies',
+            'chaise.alerts',
+            'chaise.filters',
+            'chaise.inputs',
+            'chaise.utils',
+            'ermrestjs',
+            'ui.bootstrap',
+            'chaise.errors',
+            'chaise.navbar'
+          ])
+
+            .config(['$cookiesProvider', function ($cookiesProvider) {
+                $cookiesProvider.defaults.path = '/';
+            }])
+            .factory('LineplotUtils', ['AlertsService', 'dataFormats', 'Session', 'UriUtils', '$http', '$rootScope', function (AlertsService, dataFormats, Session, UriUtils, $http, $rootScope) {
+                return {
+                    getData: function (timestamp) {
+                        var baseUri = "https://prisms.isrd.isi.edu/ermrest/catalog/1/attribute/";
+
+                        var plots = {
+                          data: []
+                        };
+                        var tracesComplete = 0;
+                        lineplotConfig.traces.forEach(function (trace) {
+                            var uriWithFilters = baseUri + trace.path + "/subject_id=" + $rootScope.subject_id + "&recorded_time::geq::" + UriUtils.fixedEncodeURIComponent(timestamp);
+                            if ($rootScope.duration) {
+                                var end_x = moment(timestamp).add($rootScope.duration, 'h').format(dataFormats.datetime.submission);
+                                uriWithFilters += "&recorded_time::leq::" + UriUtils.fixedEncodeURIComponent(end_x);
+                            }
+                            var uri = uriWithFilters + "/" + trace.x_col + "," + trace.y_col + "@sort(recorded_time)?limit=" + $rootScope.limit;
+                            $http.get(uri).then(function(response) {
```

Comment:
You should use the `$http` service that we have bundled in ermrestJS, [described here](https://github.com/informatics-isi-edu/deriva-webapps/wiki/Guideline-to-using-ermrestJS-error-handling-and-logging). This will ensure that the error handling in ermrestJS is used for the errors that are thrown.

For the service location, you should use the chaise config property, `ermrestLocation` if it's defined, then default to the default value of `<hostname>/ermrest`.